src/entities/pair.ts

```diff
@@ -35,11 +34,15 @@ export class Pair {
         ...PAIR_ADDRESS_CACHE,
         [tokens[0].address]: {
           ...PAIR_ADDRESS_CACHE?.[tokens[0].address],
-          [tokens[1].address]: getCreate2Address(
-            FACTORY_ADDRESS[chainId],
-            keccak256(['bytes'], [pack(['address', 'address'], [tokens[0].address, tokens[1].address])]),
-            INIT_CODE_HASH[chainId]
-          )
+          [tokens[1].address]: getContractAddress({
+            from: FACTORY_ADDRESS[chainId],
+            salt: keccak256(
+              encodePacked(['address', 'address'], [getAddress(tokens[0].address), getAddress(tokens[1].address)]),
+              'bytes'
+            ),
+            opcode: 'CREATE2',
+            bytecode: INIT_CODE_HASH[chainId]
+          })
```

Comment:
this doesn't work

Reply:
https://github.com/wagmi-dev/viem/blob/main/src/utils/address/getContractAddress.ts